luigi/tasks/genome_mapping/genome_mapping_tasks.py

```diff
@@ -0,0 +1,395 @@
+            for chromosome in GetChromosomeList(species=self.species, division=self.division).output():
+                yield BlatJob(fasta_input=chunk.path,
+                              species=self.species,
+                              division=self.division,
+                              chromosome=DownloadChromosome(species=self.species, chromosome=chromosome, division=self.division).output().path,
+                              taxid=self.taxid)
+
+
+class SpeciesBlatJob(luigi.Task):
+    taxid = luigi.IntParameter()
+    species = luigi.Parameter()
+    division = luigi.Parameter()
+
+    def requires(self):
+        chromosomes = []
+        for chromosome in GetChromosomeList(species=self.species, division=self.division).output():
+            chromosomes.append(DownloadChromosome(species=self.species,
+                                                  chromosome=chromosome,
+                                                  division=self.division))
+        return {
+            'chunks': CleanSplitFasta(species=self.species, taxid=self.taxid),
+            'chromosomes': chromosomes,
+        }
+
+    def run(self):
+        for chunk in self.input()['chunks']:
+            for chromosome in self.input()['chromosomes']:
+                yield BlatJob(fasta_input=chunk.path,
+                              chromosome=chromosome.path,
+                              species=self.species,
+                              division=self.division,
+                              taxid=self.taxid)
+
+    def output(self):
+        psl_files = []
+        for chunk in self.input()['chunks']:
+            for chromosome in self.input()['chromosomes']:
+                psl_files.append(BlatJob(fasta_input=chunk.path,
+                                         chromosome=chromosome.path,
+                                         species=self.species,
+                                         division=self.division,
+                                         taxid=self.taxid).output())
+        return psl_files
+
+
+class ParseBlatOutput(luigi.Task):
+    """
+    Run a shell script to transform psl output produced by blat into tsv files
+    that can be loaded into the database.
+    """
+    assembly_id = luigi.Parameter()
+    species = luigi.Parameter()
+    taxid = luigi.IntParameter()
+    division = luigi.Parameter()
+
+    def get_blat_output(self):
+        return genome_mapping().blat_output(self.species)
+
+    def requires(self):
+        return SpeciesBlatJob(taxid=self.taxid,
+                              species=self.species,
+                              division=self.division)
+
+    def output(self):
+        return {
+            'exact': luigi.LocalTarget(os.path.join(self.get_blat_output(), 'mapping.tsv')),
+            'inexact': luigi.LocalTarget(os.path.join(self.get_blat_output(), 'mapping-inexact.tsv')),
+        }
+
+    def run(self):
+        cmd = 'source scripts/psl2tsv.sh {} {}'.format(self.get_blat_output(),
```

Comment:
biopython can parse blat files (http://biopython.org/wiki/SearchIO) and I generally prefer that over some long shell script.